src/ice.c

```diff
@@ -52,6 +52,21 @@ static int parse_sdp_line(const char *line, ice_description_t *description) {
 	}
 	ice_candidate_t candidate;
 	if (ice_parse_candidate_sdp(line, &candidate) == 0) {
+		bool is_tcp = strcmp(candidate.transport, "TCP") == 0;
+		if (is_tcp && !ice_tcp_enabled) {
+			JLOG_DEBUG("ICE-TCP is disabled ignoring Candidate: %s", line);
+			return ICE_PARSE_IGNORED;
+		}
+
+		if (is_tcp) {
+			for (int i = 0; i < description->candidates_count; ++i) {
+				if (strcmp(description->candidates[i].transport, "TCP") == 0) {
+					JLOG_DEBUG("Only one ICE-TCP remote candidate is supported ignoring Candidate: %s", line);
+					return ICE_PARSE_IGNORED;
```

Comment:
It looks like there is still confusion between the types of TCP candidates, if the remote has a passive and an active candidate it will ignore the second one, irrelevant of its type. I think it would be better to move this check when creating entries for candidates in `agent_add_candidate_pair()`.

Reply:
I believe this might be too late. The candidates are already copied into `agent->remote.candidates_count` (etc..)

So we would have to tag/mark candidates as `we parsed and copied this into memory, but did not actually use them`

what do you think?

Reply:
I think it's fine, there is no issue with keeping the candidates even if there are not used later. You don't even need to mark the candidates, only no entries should be created for them.

Reply:
Fixed! Moved this check into `agent_add_candidate_pair`